web/modules/custom/dpl_favorites_list/dpl_favorites_list.module

```diff
@@ -3,6 +3,61 @@
 /**
  * @file
  * Dpl_favorites_list drupal module file.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\dpl_favorites_list\DplFavoritesListSettings;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function dpl_favorites_list_form_dpl_library_agency_list_size_settings_form_alter(array &$form, FormStateInterface $form_state): void {
+  $config = \Drupal::service('config.factory')
+    ->getEditable('dpl_library_agency.list_size_settings');
+
+  $form['favorites_list_size_settings'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Favorites list size settings', [], ['context' => 'List size (settings)']),
+    '#description' => t('The number of items to display in the favorites list.', [], ['context' => 'List size (settings)']),
+    '#tree' => FALSE,
+  ];
+
+  $form['favorites_list_size_settings']['favorites_list_size_desktop'] = [
+    '#type' => 'number',
+    '#title' => t('Favorites list size on desktop', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('favorites_list_size_desktop') ?? DplFavoritesListSettings::FAVORITES_LIST_SIZE_DESKTOP,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['favorites_list_size_settings']['favorites_list_size_mobile'] = [
+    '#type' => 'number',
+    '#title' => t('Favorites list size on mobile', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('favorites_list_size_mobile') ?? DplFavoritesListSettings::FAVORITES_LIST_SIZE_MOBILE,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['#submit'][] = 'dpl_favorites_list_handler_submit';
+}
+
+/**
+ * Custom submit callback.
  *
- * Renders the favorites list.
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
  */
+function dpl_favorites_list_handler_submit(array &$form, FormStateInterface $form_state): void {
+  \Drupal::service('config.factory')
```

Comment:
You should use config service here instead.

Reply:
I tried implementing the use of config service here, but only with limited success.
Config service will not let me use `->set()` function, as the config is immutable.

Do you have another suggestion?